packages/unmock-core/src/service/validator.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Contains logic about validating a state request for a service.
+ */
+
+import {
+  isSchema,
+  MediaType,
+  Operation,
+  Response,
+  Schema,
+  UnmockServiceState,
+} from "./interfaces";
+
+export class StateRequestValidator {
```

Comment:
This is far from ideal design, but it will be resolved in #63 - this is simply to get things going...

Reply:
As TypeScript does not have static classes, could use `abstract` here to make it clear it's not supposed to be instantiated.

Reply:
Removed from a class anyway (again, in #63) - it's not stateful so I don't see a point in it being a class...

Reply:
We're already using classes for non-stateful things like the matcher, I think it makes the architecture clear. But ofc it does not need to be a class.

Reply:
True, I believe that's useful when there are multiple methods/functions being available. Here the end goal is just one function (with 2 more being exported for testing purposes), so I'd rather go with functions that are 'namespaced' by being in the same file.